Tests/SwiftStdlibTests/DictionaryExtensionsTests.swift

```diff
@@ -78,6 +78,16 @@ final class DictionaryExtensionsTests: XCTestCase {
         XCTAssertEqual(dict, ["testkey": "value"])
     }
 
+    func testSubscriptKeypath() {
+        var json = ["key": ["key1": ["key2": "value"]]]
+
+        XCTAssertEqual(json[path: []] as? String, nil)
```

Comment:
Is it necessary to cast `as? String`? Are any of these castings necessary?

Reply:
Yes. Because we are returning type `Any` and `Any` and `String` cannot be compared. If we returned generic type `T`, this would not be necessary.